### PR TITLE
only set CGO_ENABLED=0 for tests

### PR DESCRIPTION
**Pull request by jim-minter**
#11044 accidentally caused the openshift binary to be built with CGO_ENABLED=0 if it had not been built by the time test/extended/core.sh was run.  The openshift binary cannot run correctly with CGO_ENABLED=0 as cgo is used by kubelet.

This change centralises the setting of CGO_ENABLED to hack/common.sh for all .test executables.

I believe that this setting is wanted so that delve can be used against these executables.

### CODE REVIEW

**Comment by jim-minter** [comment]
@smarterclayton @stevekuznetsov @liggitt @marun as you were in the conversation on #11044

**Comment by bparees** [comment]
defer to @smarterclayton and @stevekuznetsov

**Comment by smarterclayton** [comment]
Looks fine to me [test]

**Comment by jim-minter** [comment]
flake #12558
[test]

**Comment by jim-minter** [comment]
[testextended][extended:all]

**Comment by jim-minter** [comment]
flakes #12362 #12499 #12500 12501

**Comment by jim-minter** [comment]
[test]

**Comment by jim-minter** [comment]
@smarterclayton please merge if happy

File: hack/build-cross.sh [hunk 1]
```diff
@@ -9,7 +9,9 @@ host_platform="$(os::build::host_platform)"
 # Set build tags for these binaries
 readonly OS_GOFLAGS_TAGS="include_gcs include_oss"
 readonly OS_GOFLAGS_TAGS_LINUX_PPC64LE="gssapi"
+readonly OS_GOFLAGS_TAGS_TEST_LINUX_PPC64LE=""
```
Comment (smarterclayton) [inline hack/build-cross.sh]: You shouldn't have to do this - empty strings = no-op

Comment (jim-minter) [inline hack/build-cross.sh]: fixed

**Comment by smarterclayton** [comment]
LGTM [merge]

File: hack/common.sh [hunk 2]
```diff
@@ -213,6 +211,7 @@ os::build::internal::build_binaries() {
       fi
 
       local platform_gotags_envvar=OS_GOFLAGS_TAGS_$(echo ${platform} | tr '[:lower:]/' '[:upper:]_')
+      local platform_gotags_test_envvar=OS_GOFLAGS_TAGS_TEST_$(echo ${platform} | tr '[:lower:]/' '[:upper:]_')
```
Comment (stevekuznetsov) [inline hack/common.sh]: Why was this change necessary? It looks like it's going to break anyone who was using `OS_GOFLAGS_TAGS` to pass to test builds

Comment (jim-minter) [inline hack/common.sh]: I don't believe it does break that.  The change stops automatically setting the gssapi tag, which is incompatible with CGO_ENABLED=0.

Comment (stevekuznetsov) [inline hack/common.sh]: If I were passing the platform-specific `OS_GOFLAGS_TAGS_LINUX` today, that would get used in the build of tests. You've changed that so I need to pass `OS_GOFLAGS_TAGS_TEST_LINUX` for that. This is a breaking change. I'm not sure we have a good way to identify if someone downstream will be broken by such a change.

Comment (smarterclayton) [inline hack/common.sh]: @liggitt is the main potential impact with gssapi

In general test tags should be explicit - if we need gssapi in integration, we need to call that out (which means it would fail on Macs, which it doesn't).

Comment (liggitt) [inline hack/common.sh]: gssapi is only for the kubectl build. We have an extended test that will break shortly if this broke something
